src/drawerbase.js

```diff
@@ -0,0 +1,291 @@
+        // This issue only affects our canvas renderer, but we do it always for consistency.
+        // Note that this means overlays you want to be rtl need to be explicitly set to rtl.
+        this.container.dir = 'ltr';
+
+        // the first time this.canvas is accessed, implementations will create it
+        this.canvas.style.width     = "100%";
+        this.canvas.style.height    = "100%";
+        this.canvas.style.position  = "absolute";
+        $.setElementOpacity( this.canvas, this.viewer.opacity, true ); // set
+
+        // Allow pointer events to pass through the canvas element so implicit
+        //   pointer capture works on touch devices
+        $.setElementPointerEventsNone( this.canvas );
+        $.setElementTouchActionNone( this.canvas );
+
+        // explicit left-align
+        this.container.style.textAlign = "left";
+        this.container.appendChild( this.canvas );
+
+        this._checkForAPIOverrides();
+    }
+
+    get canvas(){
+        if(!this._renderingTarget){
+            this._renderingTarget = this._createDrawingElement();
+        }
+        return this._renderingTarget;
+    }
+    get element(){
+        $.console.error('Drawer.element is deprecated. Use Drawer.container instead.');
+        return this.container;
+    }
+
+    /**
+     * @property {String|undefined} type What type of drawer this is. Implementations should override this property.
+     */
+    getType(){
+        return undefined;
+    }
+
+    /**
+     * @returns {Boolean} whether the drawer implementation is supported by the browser
+     */
+    static isSupported() {
+        $.console.error('Drawer.isSupported must be implemented by child class');
+    }
+
+    /**
+     * create the HTML element (e.g. canvas, div) that the image will be drawn into
+     * @returns {Element} the element to draw into
+     */
+    _createDrawingElement() {
+        $.console.error('Drawer._createDrawingElement must be implemented by child class');
+        return null;
+    }
+
+    /**
+     * @param {Array} tiledImages - An array of TiledImages that are ready to be drawn
+     */
+    draw(tiledImages) {
+        $.console.error('Drawer.draw must be implemented by child class');
+    }
+
+    /**
+     * @returns {Boolean} True if rotation is supported.
+     */
+    canRotate() {
+        $.console.error('Drawer.canRotate must be implemented by child class');
+    }
+
+    /**
+     * Destroy the drawer (unload current loaded tiles)
+     */
+    destroy() {
+        $.console.error('Drawer.destroy must be implemented by child class');
+    }
+
+    /**
+     * @returns {Boolean} Whether this drawer requires enforcing minimum tile overlap to avoid showing seams.
+     */
+    minimumOverlapRequired() {
+        return false;
+     }
+
+
+    /**
+     * Turns image smoothing on or off for this viewer. Note: Ignored in some (especially older) browsers that do not support this property.
```

Comment:
Maybe the comment should mention that it's only for the CanvasDrawer?

Reply:
The WebGLDrawer also supports this, it's only HTML drawer that doesn't.

Reply:
Interesting! Is that built into WebGL or something you're doing in the drawer? I guess I'll see when I get there...